File: 118PascalsTriangle.py
class Solution:
    def generate(self, numRows):
        """

        :param numRows: int
        :return: List[List[int]]
        """
        triangle = []
        for row_num in range(numRows):
            row = [0 for _ in range(row_num+1)]
            row[0] = 1
            row[-1] = 1

            for col_num in range(1, len(row)-1):
                row[col_num] = triangle[row_num-1][col_num-1] + triangle[row_num-1][col_num]
            triangle.append(row)
        return triangle



# Expected "11"
N = 1
s = Solution()
print(s.generate(N))

File: 118PascalsTriangle.cpp
#include<iostream>
#include<vector>
#include<cassert>

using namespace std;

class Solution {
public:
	vector<vector<int>> generate(int numRows) {
		vector<vector<int>> res(numRows, vector<int>());
		for (int i = 0; i < numRows; i++)
		{
			res[i].resize(i + 1, 1);
			for (int j = 1; j < i; j++)
			{
				res[i][j] = res[i - 1][j - 1] + res[i - 1][j];
			}
		}
		return res;
	}
};

int main()
{
	Solution sol;
	int numRows = 4;
	vector<vector<int>> res;
	res.push_back({ 1 });
	res.push_back({ 1, 1 });
	res.push_back({ 1, 2, 1 });
	res.push_back({ 1, 3, 3, 1 });

	vector<vector<int>> solRes = sol.generate(numRows);
	for (int i = 0; i < numRows; i++)
	{
		for (int j = 0; j < res[i].size(); j++)
		{
			assert(res[i][j] == solRes[i][j]);
		}
	}
	cout << "All test cases passed!\n";
}